pilot/pkg/model/context.go

```diff
@@ -492,6 +492,19 @@ func compareVersion(ov, nv int) int {
 
 var NodeTypes = [...]NodeType{SidecarProxy, Router, Waypoint, Ztunnel}
 
+// Key returns the key of Proxy based on its addresses, have the same logic as Key() of IstioEndpoint
+func (node *Proxy) Key() string {
```

Comment:
How about "FirstIPOrNil" or something less generic ? 

The IP is one of the worst things to use as a key...